src/lib/server/module.c

```diff
@@ -966,7 +966,7 @@ module_instance_t *module_by_name_and_method(module_method_t *method, rlm_compon
 	 *
 	 *	Loop over the method names, seeing if we have a match.
 	 */
-	len = q - p;
+	len = strlen(p);
```

Comment:
I don't think this is correct.  The name can be `module.method1.method2`.  So doing `strlen(p)` here means it's comparing `method1.method2`, and not just `method1`.